ArduCopter/heli.cpp

```diff
@@ -192,13 +192,20 @@ void Copter::heli_update_autorotation()
         helispdhgtctrl->set_enable(0);
     }
 
-    if (!ap.land_complete && !motors->get_interlock() && arot->is_enable()) {
-        heli_flags.in_autorotation = true;
+    if (!ap.land_complete && !motors->get_interlock()) {
         // Check if an auto collective mode is being used
-        if (!flightmode->has_manual_throttle()){
+        if (!flightmode->has_manual_throttle() && arot->is_enable()){
+            //autonomous autorotation
+            heli_flags.in_autorotation = true;
             set_mode(Mode::Number::AUTOROTATE, MODE_REASON_AUTO_AUTOROTATION);
+        } else if (!flightmode->has_manual_throttle() && !arot->is_enable()){
+            //autorotation flight mode is not enabled
+            heli_flags.in_autorotation = false;
+        } else {
+            //autorotation in for manual throttle modes
+            heli_flags.in_autorotation = false;
```

Comment:
what other option is there for the inner "if" block?  The "else if" condition can be covered by the "else".  So, I think you could get rid of the "else if" statement and just leave the "else" statement.  then for the outer "if" block, the "else" statement covers everything else.

Reply:
The reason for putting the else if statement was to allow for the manual-throttle autorotation case.  I set in autorotation to false for now as I couldn't remember the extent/affects of the in_autorotation flag.  In hindsight this is pretty confusing.  I am happy to get rid of it if you would prefer or I can set the in_autorotation flag to true.

Reply:
I appreciate that we have more to do to get the manual autorotation case working so this was more of a space saver

Reply:
I'm not so sure that we have much to do with manual autorotation other than enabling the bailout functionality.  I think I used the in_autorotation flag only to signify to the motors class to use the autorotation limits of the collective when we had autorotation collective limts that were different from the normal collective limits.  So if we never put that back in, I don't see the need to tell the motors class that we are in autorotation, just bailout.  I guess let's remove it for the time being.  We'll put it back in if the need arises.

Reply:
I have removed it.